library/src/compute/calc.rs

```diff
@@ -404,6 +408,170 @@ pub fn log(
     Value::Float(result)
 }
 
+/// Calculates the product of a range of numbers. Used to calculate permutations.
+/// Returns None if the result is larger than `u64::MAX`
+fn factorial_range(start: u64, end: u64) -> Option<u64> {
+    // By convention
+    if end + 1 < start {
+        return Some(0);
+    }
+
+    let mut count: u64 = 1;
+    let real_start: u64 = cmp::max(1, start);
+
+    for i in real_start..=end {
+        match count.checked_mul(i) {
+            None => return None,
+            Some(s) => count = s,
+        };
+    }
+    Some(count)
+}
+
+/// Calculates the factorial of an integer.
+/// Returns `None` if the result is larger than `u64::MAX`
+fn factorial(number: u64) -> Option<u64> {
```

Comment:
I'd say you can just use `factorial_range` directly on the one call site of this function. It's not really worth it in my opinion.